cpp/src/io/parquet/parquet_gpu.cuh

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "parquet_gpu.hpp"
+
+#include <cudf/lists/lists_column_device_view.cuh>
+#include <cudf/types.hpp>
+
+#include <cuco/static_map.cuh>
+
+namespace cudf::io::parquet::gpu {
+
+auto constexpr KEY_SENTINEL   = size_type{-1};
+auto constexpr VALUE_SENTINEL = size_type{-1};
+
+using map_type = cuco::static_map<size_type, size_type>;
+
+/**
+ * @brief The alias of `map_type::pair_atomic_type` class.
+ *
+ * Declare this struct by trivial subclassing instead of type aliasing so we can have forward
+ * declaration of this struct somewhere else.
+ */
+struct slot_type : public map_type::pair_atomic_type {
+};
```

Comment:
Maybe moving to `.hpp` since no device code is involved.

Reply:
This struct inherits `map_type` which is from `cuco/static_map.cuh` that requires to be in a `cuh` file.

Reply:
Dang... you are right. We have an issue to track this actually https://github.com/NVIDIA/cuCollections/issues/232